src/server/config/index.js

```diff
@@ -7,14 +7,24 @@ var configPath = process.env.CONFIG_PATH || path.join(__dirname, 'kibana.yml');
 var kibana = yaml.safeLoad(fs.readFileSync(configPath, 'utf8'));
 var env = process.env.NODE_ENV || 'development';
 
+function checkPath(path) {
+  try {
+    fs.statSync(path);
+    return true;
+  } catch (err) {
+    return false;
+  }
+}
+
 // Check if the local public folder is present. This means we are running in
 // the NPM module. If it's not there then we are running in the git root.
 var public_folder = path.resolve(__dirname, '..', 'public');
-try {
-  fs.statSync(public_folder);
-} catch (err) {
-  public_folder = path.resolve(__dirname, '..', '..', 'kibana');
-}
+if (!checkPath(public_folder)) public_folder = path.resolve(__dirname, '..', '..', 'kibana');
+
+// Check to see if htpasswd file exists in the root directory otherwise set it to false
+var htpasswdPath = path.resolve(__dirname, '..', '.htpasswd');
```

Comment:
This is the server (written in Node.js) portion of the code. The .htpasswd file is not accessible to the client.